### ISSUE TITLE
[a11y]: Disabled icon button doesn't show the tooltip

### ISSUE THREAD

**Issue by kromakollision** [carbon-design-system/carbon#16475]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.55.0

### React version

18.2.0

### Automated testing tool and ruleset

none

### Description

An icon-only button has a mandatory tooltip, for accessiblity. However, this tooltip doesn't show up when the button is disabled.

### Reproduction/example

https://stackblitz.com/edit/github-9lactv?file=src%2FApp.jsx

### Steps to reproduce

Add `disabled` to `<Button />`

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
This is working as intended as you can't focus on a disabled button so there would be no way to activate a tooltip. 

> Disabled buttons are intentionally removed from the focus order, according to the spec. Therefore, any information revealed on hover cannot be surfaced on focus, resulting in an accessibility failure.

Related issue with more details https://github.com/carbon-design-system/carbon-website/issues/3758
